stream/client.go

```diff
@@ -82,6 +82,9 @@ func (c *Client) StreamSearch(ctx context.Context, q query.Q, opts *zoekt.Search
 		reply := &searchReply{}
 		err := dec.Decode(reply)
 		if err != nil {
+			if ctxErr, ok := stringToContextError(err.Error()); ok {
```

Comment:
Converting error to a string to check for context canceled feels dangerous. Can't you use `errors.Is`?

Reply:
You are right!  For some reason I was convinced that `dec.Decode` doesn't return a proper context error because gob doesn't support encoding/decoding errors.